### DISCUSSION TITLE
Derivate parameters

### DISCUSSION THREAD

**Discussion by MarcoArtiano** [xcompact3d/Incompact3d#239] [Q&A] [answered]
Hello, I'm trying to implement a different formulation for the discretization of the convective scheme, I was wondering why for the derivatives there are different parameters, for example, line 117-122 of transeq.f90

`   
    call derx (td1,ta1,di1,sx,ffxp,fsxp,fwxp,xsize(1),xsize(2),xsize(3),1,ubcx*ubcx)    # d/dx (uu)
    call derx (te1,tb1,di1,sx,ffx,fsx,fwx,xsize(1),xsize(2),xsize(3),0,ubcx*ubcy)          # d/dx (vv)
    call derx (tf1,tc1,di1,sx,ffx,fsx,fwx,xsize(1),xsize(2),xsize(3),0,ubcx*ubcz)           # d/dx (ww)
    call derx (ta1,ux1,di1,sx,ffx,fsx,fwx,xsize(1),xsize(2),xsize(3),0,ubcx)                  # d/dx (u)
    call derx (tb1,uy1,di1,sx,ffxp,fsxp,fwxp,xsize(1),xsize(2),xsize(3),1,ubcy)            # d/dx (v) 
    call derx (tc1,uz1,di1,sx,ffxp,fsxp,fwxp,xsize(1),xsize(2),xsize(3),1,ubcz)            # d/dx (w)

`

where on the side I wrote "their meaning", if I'm correct.

Now, for example, I would like to change the x advective component of the non linear term with udu/dx, vdv/dx, wdw/dx. In order to do that should I modifiy some of the parameters appearing in the call function derx? Why for some of them there's 1 or 0 at the end, and what's the difference between ffxp, and ffx? 

Because I simply used ta1*ux1 + tb1*uy1 + tc1*uz1 and of course I modified the y and z component according to the new formulation that I'm trying (this is only an example for the x component)

Thank you in advance!

**Comment by slaizet** [comment, marked as answer]
Dear Marco,
When using ncl=1, you can have symmetry and anti-symmetry boundary conditions (npaire=1=symmetry and npaire=0=anti-symmetry). For each case, the set of coefficients for the derivatives is slightly different hence the use of the arrays with "p" at the end. If you are not using ncl=1, then do not worry about this. If you want to implement a new formulation for the convective terms, simply follow the same rules used in the current implementation (for instance, d/dx (uu) --> npaire=1, d/dx (ww) --> npaire=0).
All the best,
Sylvain
